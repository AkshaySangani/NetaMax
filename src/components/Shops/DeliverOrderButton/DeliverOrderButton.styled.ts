import styled from "styled-components";

import { Box, Button } from "@chakra-ui/react";

export const DeliverButton = styled(Box)`
  background: #ffff00;
  align-content: center;
  align-items: center;
  align-self: center;
  display: flex;
  margin-left: 6px;
  margin-right: 6px;
  justify-content: center;
  border-radius: 16px;
  max-width: 482px;
  margin-right: auto;
  margin-left: auto;
  opacity: 1 !important;
`;

export const ButtonNoOp = styled(Button)`
  opacity: 1 !important;
`;
