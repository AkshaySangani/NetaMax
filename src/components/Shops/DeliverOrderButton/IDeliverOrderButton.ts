export interface IDeveliveryOrderButtonProps {
  /**
   * The onOpen function to show the pick up auth token flow
   * @type {() => void}
   */
  onOpen: () => void;
}
