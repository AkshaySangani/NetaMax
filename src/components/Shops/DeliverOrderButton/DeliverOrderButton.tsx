import { ReactElement } from "react";

import IconDeliveryBasket from "styled/icons/Shops/PickUp/DeliveryBasket";

import { Box, Center, Flex, Text } from "@chakra-ui/react";

import { DeliverButton } from "./DeliverOrderButton.styled";
import { IDeliverOrderButtonProps } from "./IDeliverOrderButtonProps";

/**
 * The Deliver Order Button
 * @param {IDeliverOrderButtonProps} props for DeliverOrderButton component
 * @returns {ReactElement} The Deliver Order Button
 */
export const DeliverOrderButton = (props: IDeliverOrderButtonProps): ReactElement => {
  const { onOpen } = props;

  return (
    <Box p="16px" bottom="0" left="0" width="100%">
      <DeliverButton
        onClick={onOpen}
        height="48px"
        display="flex"
        alignItems="center"
        cursor="pointer"
      >
        <Center>
          <Flex justifyContent="space-between" alignItems="center" width="100%" textAlign="center">
            <Box mr="19px">
              <IconDeliveryBasket />
            </Box>
            <Text fontWeight="600">Entregar pedido</Text>
          </Flex>
        </Center>
      </DeliverButton>
    </Box>
  );
};
