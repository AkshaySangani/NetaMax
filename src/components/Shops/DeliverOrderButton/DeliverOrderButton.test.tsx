import { render, cleanup, screen, fireEvent } from "@testing-library/react";
import { DeliverOrderButton } from "./DeliverOrderButton";
import { IDeliverOrderButtonProps } from "./IDeliverOrderButtonProps";

describe("DeliverOrderButton component", () => {
  let mock: IDeliverOrderButtonProps;
  const onOpen = jest.fn();

  beforeEach(() => {
    mock = {
      onOpen,
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("DeliverOrderButton should be render correctly", () => {
    const { container } = render(<DeliverOrderButton {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("Entregar pedido button should work properly", () => {
    render(<DeliverOrderButton {...mock} />);
    fireEvent.click(screen.getByText("Entregar pedido"));
    expect(onOpen).toHaveBeenCalled();
  });
});
