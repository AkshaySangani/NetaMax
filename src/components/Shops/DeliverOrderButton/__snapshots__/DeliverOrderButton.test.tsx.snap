// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliverOrderButton component DeliverOrderButton should be render correctly 1`] = `
<div>
  <div
    class="css-5tgvyw"
  >
    <div
      class="sc-bdvvtL gVSGcd css-14gea8e"
    >
      <div
        class="css-gmuwbf"
      >
        <div
          class="css-1blbj9p"
        >
          <div
            class="css-1htrffh"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 20C17 19.4477 17.4477 19 18 19C18.5523 19 19 19.4477 19 20C19 20.5523 18.5523 21 18 21C17.4477 21 17 20.5523 17 20Z"
                stroke="#353535"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M7 20C7 19.4477 7.44772 19 8 19C8.55228 19 9 19.4477 9 20C9 20.5523 8.55228 21 8 21C7.44772 21 7 20.5523 7 20Z"
                stroke="#353535"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M3 3H3.76772C4.24998 3 4.66354 3.3442 4.7511 3.81845L7 16"
                stroke="#353535"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M5.5 6H19.7429C20.386 6 20.8619 6.59824 20.7173 7.22486L18.8712 15.2249C18.7665 15.6786 18.3625 16 17.8968 16H7"
                stroke="#353535"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <p
            class="chakra-text css-35ezg3"
          >
            Entregar pedido
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
